### PR TITLE
[action] fix reset_simulator_contents to work with Xcode 9 and later

### LINKED ISSUE #12731: reset_simulators no longer works
- Reported in #12284
- Also reported by @danielbowden (in person)

@danielbowden's workaround solution:
```rb
private_lane :erase_simulator do
    simulators = FastlaneCore::DeviceManager.simulators('iOS')
    # Find latest simulator for test_device
    device = simulators.select { |s| s.name == test_device }
                 .sort_by { |s| Gem::Version.create(s.os_version) }
                 .last

    FastlaneCore::Simulator.reset(udid: device.udid) unless device.nil?
  end
```

### PR DESCRIPTION
**Pull request by joshdholtz**
Fixes #12731

## What
- The `reset_simulator_contents` action was not implemented for Xcode9 or later
- This adds support for Xcode9 simulators
- Separated out the Xcode8 and under and the Xcode9 and over into separate methods

## What was tested?
- Xcode9
  - I tested resetting contents of all Xcode 9 simulators and when specifying a specific version
- Xcode8
  - I installed a version of Xcode8 but could not successfully reset contents due to `xcrun simctl list device` errors I wasn't able to run
  - Not sure if anybody can still run this, TBH 😊 
  - It might be worth deleting at some point 🤔

### CODE REVIEW

File: fastlane/lib/fastlane/actions/reset_simulator_contents.rb [hunk 1]
```diff
@@ -2,13 +2,31 @@ module Fastlane
...
-          UI.important("Resetting simulators currently doesn't work with Xcode 9, stay tuned as we are working to add support for all new tools.")
-          return
+          reset_at_least_xcode9(ios_versions)
+        else
+          reset_up_to_xcode8(ios_versions)
+        end
+      end
+
+      def self.reset_at_least_xcode9(ios_versions)
+        UI.verbose("Resetting simulator contents forXcode9 and later")
```
Comment (revolter) [inline fastlane/lib/fastlane/actions/reset_simulator_contents.rb]: `for Xcode 9`*

File: fastlane/lib/fastlane/actions/reset_simulator_contents.rb [hunk 2]
```diff
@@ -2,13 +2,31 @@ module Fastlane
...
+        simulators.each do |simulator|
+          FastlaneCore::Simulator.reset(udid: simulator.udid)
         end
+        UI.success('Simulators reset')
+      end
 
-        if params[:ios]
-          params[:ios].each do |os_version|
+      def self.reset_up_to_xcode8(ios_versions)
+        UI.verbose("Resetting simulator contents for Xcode8 and earlier")
```
Comment (revolter) [inline fastlane/lib/fastlane/actions/reset_simulator_contents.rb]: `Xcode 8`*?

File: fastlane/lib/fastlane/actions/reset_simulator_contents.rb [hunk 3]
```diff
@@ -2,13 +2,31 @@ module Fastlane
   module Actions
     class ResetSimulatorContentsAction < Action
       def self.run(params)
+        ios_versions = params[:ios]
+
         if Helper.xcode_at_least?("9")
-          UI.important("Resetting simulators currently doesn't work with Xcode 9, stay tuned as we are working to add support for all new tools.")
```
Comment (taquitos) [inline fastlane/lib/fastlane/actions/reset_simulator_contents.rb]: 🎉

File: fastlane/lib/fastlane/actions/reset_simulator_contents.rb [hunk 4]
```diff
@@ -2,13 +2,31 @@ module Fastlane
...
-          return
+          reset_at_least_xcode9(ios_versions)
```
Comment (taquitos) [inline fastlane/lib/fastlane/actions/reset_simulator_contents.rb]: `reset_at_least_xcode9` -> `reset_xcode9_and_higher`

**Review by taquitos — APPROVED** [review]
Doesn't seem like it should break anything, one naming nit, and then 🎈:goat: it

**Pull request merged**